utils/idmloader/IDM.md

```diff
@@ -0,0 +1,173 @@
+# MODFLOW 6 Input Data Model developer resource
+
+This document intends to describe, from a development perspective, what the MODFLOW 6 Input Data Model (IDM) is and how to work with it. As the Input Data Model is an active area of development this document should be expected to change frequently and will be updated to track with current state.
+
+* [Overview](#overview)
+* [Terminology](#terminology)
+* [IDM roadmap](#idm-roadmap)
+* [Package Update Status](#package-update-status)
+* [Package Update Process](#package-update-process)
+
+## Overview
+The MODFLOW 6 Input Data Model is intended to provide a common entry point for simulation input data, which can be referred to as the "input context". It is built upon the MODFLOW 6 memory management infrastructure to define variable access paths that are scoped to a particular component/subcomponent. Input variables themselves are ultimately described by MODFLOW 6 definition (dfn) files.
```

Comment:
```suggestion
The MODFLOW 6 IDM is intended to provide a common entry point for simulation input data.  The approach is to read the input data first into memory and store the input data in the "input context."  Reading the simulation input into memory is performed by IDM load routines.  Model packages are then refactored so that they obtain their required information from the input context, rather than directly from the input files.  We refer to the process as "sourcing."  A model package "sources" its input from the input context.  The advantage here is that other readers can be implemented to load information into the input context, and then the model packages should work without modification.   This approach is built upon the MODFLOW 6 memory management infrastructure to define variable access paths that are scoped to a particular component/subcomponent. Input variables themselves are ultimately described by MODFLOW 6 definition (dfn) files.
```